api/src/org/apache/cloudstack/context/CallContext.java

```diff
@@ -90,7 +90,13 @@ public void putContextParameter(Object key, Object value) {
     }
 
     public Object getContextParameter(Object key) {
-        return context.get(key);
+        Object value = context.get(key);
+        //check if the value is present in the toString value of the key
+        //due to a bug in the way we update the key by serializing and deserializing, it sometimes gets toString value of the key. @see com.cloud.api.ApiAsyncJobDispatcher#runJob
+        if(value == null ) {
+            value = context.get(key.toString());
```

Comment:
key cannot be null. There is a not null check at the only place where it is called ActionEventUtils.publishOnEventBus(). updated the javadoc